model/addons_mgmt/v1/addon_config_type.model

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright (c) 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Representation of an addon config.
+// The attributes under it are to be used by the addon once its installed in the cluster.
+struct AddonConfig {
+    // List of environment variables for the addon
+    AddOnEnvironmentVariables []AddonEnvironmentVariable
+
+    // List of secret propagations for the addon
+    AddOnSecretPropagations []AddonSecretPropagation
```

Comment:
actually I'm wondering if we even need Addon as part of these variable names: since it's part of the config for the addon isn't it a given that the environment variables and secrets are for the addon?

Reply:
we are kind of stuck with this, as it is currently in the API, changing it will break it. Lets discuss this on next weeks engineering call and come up with a decision on it. We have time to make changes here before we implement into the clients.